File: lib/features/areas/models/create_area_comun_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'create_area_comun_request.g.dart';

@JsonSerializable()
class CreateAreaComunRequest {
  final String nombre;
  final String? descripcion;
  final int capacidad;
  final String horarioInicio;
  final String horarioFin;
  final int duracionBloqueMinutos;
  final int maxReservasMesPorUsuario;
  final int anticipacionMinimaHoras;
  final int anticipacionMaximaDias;
  final bool activa;

  CreateAreaComunRequest({
    required this.nombre,
    this.descripcion,
    required this.capacidad,
    required this.horarioInicio,
    required this.horarioFin,
    required this.duracionBloqueMinutos,
    required this.maxReservasMesPorUsuario,
    required this.anticipacionMinimaHoras,
    required this.anticipacionMaximaDias,
    required this.activa,
  });

  factory CreateAreaComunRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateAreaComunRequestFromJson(json);
  Map<String, dynamic> toJson() => _$CreateAreaComunRequestToJson(this);
}

File: backend/src/main/java/com/condos/area/dto/CreateAreaComunRequest.java
package com.condos.area.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class CreateAreaComunRequest {

    @NotBlank
    private String nombre;

    private String descripcion;

    @NotNull
    @Min(1)
    private Integer capacidad;

    @NotBlank
    private String horarioInicio;

    @NotBlank
    private String horarioFin;

    @NotNull
    @Min(0)
    private Integer duracionBloqueMinutos;  // 0 = todo el día (un solo bloque)

    @NotNull
    @Min(1)
    private Integer maxReservasMesPorUsuario;

    @NotNull
    @Min(0)
    private Integer anticipacionMinimaHoras;  // 0 = sin restricción mínima

    @NotNull
    @Min(0)
    private Integer anticipacionMaximaDias;   // 0 = sin restricción máxima

    private boolean activa = true;
}
